### PR TITLE
Fixed a bug where users would lose role associations on update and not show up in ART

### PR DESCRIPTION
**Pull request by hbgwhite**
Also updated tests to adopt new logic.

### CODE REVIEW

File: persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java [hunk 1]
```diff
@@ -135,13 +129,14 @@ private boolean isAcceptableRole(User.RoleAssociation roleAssociation, boolean a
         return acceptableRole;
     }
 
-    private String getStateAbbreviationFromMatchingId(List<Sb11Entity> entities, String entityId) {
+    private User.RoleAssociation decorateWithStateAbbreviationAndMongoId(List<Sb11Entity> entities, String entityId, User.RoleAssociation roleAssociation) {
         for (Sb11Entity entity: entities) {
             if(entity.getEntityId().equals(entityId)) {
-                return entity.getStateAbbreviation();
+                roleAssociation.setStateAbbreviation(entity.getStateAbbreviation());
+                roleAssociation.setAssociatedEntityMongoId(entity.getId());
```
Comment (millerts) [inline persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java]: Suggest breaking out of the loop here after you've set the values

Comment (millerts) [inline persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java]: Also, it looks like your now associating the MongoId which wasn't being done prior to this change.  I assume that is fine but just calling out the difference if that wasn't an intended change.

File: persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java [hunk 2]
```diff
@@ -101,29 +101,23 @@ private boolean isAcceptableRole(User.RoleAssociation roleAssociation, boolean a
                     case STATE:
                         List<Sb11Entity> stateEntities = entityLocatorService.findEntities(new HashMap<String, String[]>(), FormatType.STATE);
                         List<Sb11Entity> mergedStateEntites =  (List<Sb11Entity>)(List<?>) stateService.mergeWithNonAccessStatesInTenancy(((List<StateEntity>)(List<?>)stateEntities));
-                        String stateAbbreviation = getStateAbbreviationFromMatchingId(mergedStateEntites, roleAssociation.getAssociatedEntityId());
-                        if(stateAbbreviation != null) {
+                        roleAssociation = decorateWithStateAbbreviationAndMongoId(mergedStateEntites, roleAssociation.getAssociatedEntityId(), roleAssociation);
```
Comment (jtreuting) [inline persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java]: Do you need to pass in the associated entity id?  Looks like you can just pass in the roleAssocation and get that data from within the method.

File: persistence/src/test/java/org/opentestsystem/delivery/testreg/external/ExternalUserServiceTest.java [hunk 3]
```diff
@@ -173,8 +176,10 @@ public void updateTestSuccessful() {
 
         List<Sb11Entity> locatedEntities = new ArrayList<>();
         Sb11Entity entity = new StateEntity("CA", "STATE");
+        entity.setId(UUID.randomUUID().toString());
         locatedEntities.add(entity);
         Sb11Entity entity1 = new StateEntity("DISTRICT9", "DISTRICT");
+        entity1.setId(UUID.randomUUID().toString());
```
Comment (jtreuting) [inline persistence/src/test/java/org/opentestsystem/delivery/testreg/external/ExternalUserServiceTest.java]: I don't see where these are used to confirm they got added appropriately to the roleAssociation.  Can you add them to the assert section and validate it is finding them and adding them as expected.

File: persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java (hunk 1 above)
Comment (jtreuting) [inline persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java]: @millerts that is the bug fix actually.

Comment (hbgwhite) [inline persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java]: I agree- changing it

File: persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java (hunk 2 above)
Comment (hbgwhite) [inline persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/ExternalUserServiceImpl.java]: You're right  - changing it

**Pull request merged**